File: packages/firestore/ios/Plugin/Classes/Options/GetDocumentOptions.swift
import Foundation

@objc public class GetDocumentOptions: NSObject {
    private var reference: String
    private var serverTimestampBehavior: String?

    init(reference: String, serverTimestampBehavior: String? = nil) {
        self.reference = reference
        self.serverTimestampBehavior = serverTimestampBehavior
    }

    func getReference() -> String {
        return reference
    }

    func getServerTimestampBehavior() -> String? {
        return serverTimestampBehavior
    }
}

File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/GetDocumentOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.Nullable;

public class GetDocumentOptions {

    private final String reference;

    @Nullable
    private final String serverTimestampBehavior;

    public GetDocumentOptions(String reference, @Nullable String serverTimestampBehavior) {
        this.reference = reference;
        this.serverTimestampBehavior = serverTimestampBehavior;
    }

    public String getReference() {
        return reference;
    }

    @Nullable
    public String getServerTimestampBehavior() {
        return serverTimestampBehavior;
    }
}
